hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,282 @@
+          Eigen::Affine3d first_point_of_next_segment_eigen;
+          first_point_of_next_segment_eigen = Eigen::Translation3d(first_point_of_next_segment->position.x, first_point_of_next_segment->position.y, first_point_of_next_segment->position.z);
+          descartes_core::TrajectoryPtPtr first_point_of_next_segment_descartes = makeTolerancedCartesianPoint(first_point_of_next_segment_eigen);
+          process_segment_descartes.push_back(first_point_of_next_segment_descartes);
+          std::advance(it, -1); //let the hybrid iterator come back to the current segment
+        }
+
+        // Prepend the last point of the previous segment (if the current segment is not the first segment itself) 
+        if(it != unplanned_trajectory_.begin())
+        { 
+          std::advance(it, -1);
+          auto last_point_of_previous_segment = it->waypoints_geom_msgs_.begin();
+          Eigen::Affine3d last_point_of_previous_segment_eigen;
+          last_point_of_previous_segment_eigen = Eigen::Translation3d(last_point_of_previous_segment->position.x, last_point_of_previous_segment->position.y, last_point_of_previous_segment->position.z);
+          descartes_core::TrajectoryPtPtr last_point_of_previous_segment_descartes = makeTolerancedCartesianPoint(last_point_of_previous_segment_eigen);
+          process_segment_descartes.insert(process_segment_descartes.begin(), last_point_of_previous_segment_descartes);
+          std::advance(it, 1); //let the hybrid iterator come back to the current segment
+        }
+
+        /* ===== The better solution ===== */ 
+
+        // Prepend the last point of the previous _planned_ segment, i.e. joint vals of the last point of the previous segment is the seed for the current process segment   
+        // std::vector<double> joint_values = moveit_group_.getCurrentJointValues(); //this won't work always (does it depend if the previous segment was free/descrtes?)
+        // what I mean to ask is that both moveit_group_, and descartes_model_ get updated? 
+        // Best is to just extract from the overall_robot_traj_
+        /*if(it != unplanned_trajectory_.begin())
+        {
+          // Get Joint Names. 
+          // Joint names could/should be member variables of HybridPlanner class
+          std::vector<std::string> names;
+          node_handle_.getParam("controller_joint_names", names); 
+          // Get joint valur of the last waypoint in the overall_robot_traj_
+          std::vector<double> joint_values = overall_robot_traj_->getLastWayPointPtr->getJointPositions(names);        
+          descartes_core::TimingConstraint timing_constraint = TimingConstraint(0); //TODO is this correct?
+          TrajectoryPtPtr first_point = TrajectoryPtPtr(new JointTrajectoryPt(joint_values, timing_constraint)); //TODO TOFIX? timing_constraint is mandatory in constructor. Shouldn't be. 
+          process_segment_descartes.insert(process_segment_descartes.begin(), first_point);
```

Comment:
The timing constraint is an [optional argument](https://github.com/ros-industrial-consortium/descartes/blob/hydro-devel/descartes_trajectory/include/descartes_trajectory/joint_trajectory_pt.h).